### PR TITLE
fix(accessibility): add screen reader text to all loading components

### PR DESCRIPTION
**Pull request by JasonPingAesop**
Ticket: https://aesoponline.atlassian.net/browse/PUR-559

Adding screen reader text to all loading indicator in order to meet accessibility requirement.

### CODE REVIEW

File: src/constants/accessibility.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const SCREEN_READER_TEXT = 'Loading, please wait.';
```
Comment (kevin-ho87) [inline src/constants/accessibility.ts:1]: Would this mean that the loading text will always be 'Loading, please wait.' on all sites.
Is there any localisation requirements?

**Review by MkMan — COMMENTED** [review]
Some of the components take a `copy` prop that contains the content they display. This is to facilitate different languages.

Where possible, can you please add a field for loading inside `copy` perhaps with a fallback value in English instead of hardcoding?

File: src/constants/accessibility.ts (hunk 1 above)
Comment (hassanraza0054) [reply src/constants/accessibility.ts:1]: Same question!

**Review by hassanraza0054 — COMMENTED** [review]
Just got the same question as kevin regarding translations. Curious if this requires a contentful and GQL piece of work to get those in.

**Comment by hassanraza0054** [comment]
> Some of the components take a `copy` prop that contains the content they display. This is to facilitate different languages.
> 
> Where possible, can you please add a field for loading inside `copy` perhaps with a fallback value in English instead of hardcoding?

Good catch here. Perhaps the constant declared atm can be that english fallback!

File: src/components/Audio/__snapshots__/Audio.spec.tsx.snap [hunk 2]
```diff
@@ -28,6 +28,11 @@ exports[`<Audio /> should render base component correctly 1`] = `
         data-testid="data-testid-loading"
         role="status"
       >
+        <span
+          class="srOnly"
```
Comment (cyrusxzw) [inline src/components/Audio/__snapshots__/Audio.spec.tsx.snap:32]: I suggest use 'screenReaderOnly', which is easier to read than abbreviation.

Comment (MkMan) [reply src/components/Audio/__snapshots__/Audio.spec.tsx.snap:32]: I also don't like abbreviations but I think this is caused by existing code in the Loading component https://github.com/aesop/aesop-gel/blob/main/src/components/Loading/Loading.module.css#L49

File: src/constants/accessibility.ts (hunk 1 above)
Comment (JasonPingAesop) [reply src/constants/accessibility.ts]: Good catch up guys, I have removed the hard coded pieces. Allowing parent components can accept screen reader text to pass to loading components

File: src/components/Audio/__snapshots__/Audio.spec.tsx.snap (hunk 2 above)
Comment (JasonPingAesop) [reply src/components/Audio/__snapshots__/Audio.spec.tsx.snap]: True, they are css related

**Comment by JasonPingAesop** [comment]
> Just got the same question as kevin regarding translations. Curious if this requires a contentful and GQL piece of work to get those in.

True, even in web-ui repo, most of screen reader texts are hard coded in English.

**Comment by JasonPingAesop** [comment]
> Some of the components take a `copy` prop that contains the content they display. This is to facilitate different languages.
> 
> Where possible, can you please add a field for loading inside `copy` perhaps with a fallback value in English instead of hardcoding?

Maybe not fallback in english, as the loading components use `screenReaderText` variable to conditional render this text. Otherwise it will render the text forever.

File: src/components/AddToCartButton/AddToCartButton.types.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ type AddToCartButtonProps = {
   dataTestRef: string;
   isEnabled?: boolean;
   isFullWidth?: boolean;
+  screenReaderText?: string;
```
Comment (MkMan) [inline src/components/AddToCartButton/AddToCartButton.types.ts:17]: Being content presented to the user I thought `copy` would be a good place for this value. What do you think?

Comment (JasonPingAesop) [reply src/components/AddToCartButton/AddToCartButton.types.ts]: Agree, I have moved it into copy variable.

**Review by MkMan — APPROVED** [review]

**Pull request merged**